http/src/main/resources/init-resources/azure_vm_init_script.sh

```diff
@@ -152,3 +152,29 @@ jq --null-input \
 /anaconda/envs/py38_default/bin/jupyter kernelspec list | awk 'NR>1 {print $2}' | while read line; do jq -s add $line"/kernel.json" wsenv.json > tmpkernel.json && mv tmpkernel.json $line"/kernel.json"; done
 /anaconda/envs/azureml_py38/bin/jupyter kernelspec list | awk 'NR>1 {print $2}' | while read line; do jq -s add $line"/kernel.json" wsenv.json > tmpkernel.json && mv tmpkernel.json $line"/kernel.json"; done
 /anaconda/envs/azureml_py38_PT_and_TF/bin/jupyter kernelspec list | awk 'NR>1 {print $2}' | while read line; do jq -s add $line"/kernel.json" wsenv.json > tmpkernel.json && mv tmpkernel.json $line"/kernel.json"; done
+
+log 'Formatting and mounting persistent disk...'
+
+WORK_DIRECTORY='/home/jupyter/persistent_disk'
+
+# Mount persistent disk
+# The PD should be the only `sd` disk that is not mounted yet
+AllsdDisks=($(/usr/bin/lsblk --nodeps --noheadings --output NAME --paths | grep -i "sd"))
+FreesdDisks=()
+for Disk in "${AllsdDisks[@]}"; do
+    Mounts="$(/usr/bin/lsblk --noheadings --output MOUNTPOINT "${Disk}" | grep -vE "^$")"
+    if [ "${Mounts}" == "" ]; then
+        FreesdDisks+=("${Disk}")
+    fi
+done
+
+DISK_DEVICE_ID= basename ${FreesdDisks[0]}
+
+## Only format disk is it hasn't already been formatted
+## Maybe check if the working directory exists already?
+if [ ! -d ${WORK_DIRECTORY} ] ; then
```

Comment:
Thinking more, not sure this will work as intended… `/home/jupyter/persistent_disk` will _never_ exist on the VM before you mount the disk. I think this is why on GCE Leo kept track of whether a disk was previously formatted or not, and passed a flag to the script..

Reply:
Yeah, we were thinking about that too, I am thinking about mounting the disk first then trying to check if the directory exist

Reply:
I changed the logic to trying to mount the PD directly, if the PD has a formatted partition then it has data on it and should be mountable directly. If not then it must be a new disk and we have to set it up from scratch